libs/application/template-api-modules/src/lib/modules/templates/operating-license/operatingLicense.service.ts

```diff
@@ -34,11 +39,121 @@ export class OperatingLicenseService extends BaseTemplateApiService {
     @Inject(LOGGER_PROVIDER) private logger: Logger,
     private readonly sharedTemplateAPIService: SharedTemplateApiService,
     private readonly syslumennService: SyslumennService,
+    private readonly criminalRecordService: CriminalRecordService,
+    private readonly financeService: FinanceClientService,
   ) {
     super(ApplicationTypes.OPERATING_LCENSE)
     this.s3 = new S3()
   }
 
+  async criminalRecord({
+    application,
+  }: TemplateApiModuleActionProps): Promise<{ success: boolean }> {
+    const fakeData = getValueViaPath<OperatingLicenseFakeData>(
+      application.answers,
+      'fakeData',
+    )
+    const useFakeData = fakeData?.useFakeData === YES
+
+    if (useFakeData) {
+      return fakeData?.criminalRecord === YES
+        ? Promise.resolve({ success: true })
+        : Promise.reject({
+            reason: {
+              title: coreErrorMessages.dataCollectionCriminalRecordTitle,
+              summary: coreErrorMessages.dataCollectionCriminalRecordErrorTitle,
+              hideSubmitError: true,
+            },
+            statusCode: 404,
+          })
+    }
+    try {
+      const applicantSsn =
+        application.applicantActors.length > 0
+          ? application.applicantActors[0]
+          : application.applicant
+      const hasCriminalRecord = await this.criminalRecordService.validateCriminalRecord(
+        applicantSsn,
+      )
+      if (hasCriminalRecord) {
+        throw new TemplateApiError(
+          {
+            title: coreErrorMessages.dataCollectionCriminalRecordTitle,
+            summary: coreErrorMessages.dataCollectionCriminalRecordErrorTitle,
+          },
+          400,
+        )
+      }
+
+      return { success: true }
+    } catch (e) {
+      throw new TemplateApiError(
+        {
+          title: coreErrorMessages.dataCollectionCriminalRecordTitle,
+          summary: coreErrorMessages.errorDataProvider,
+        },
+        400,
+      )
+    }
+  }
+
+  async debtLessCertificate({
+    application,
+    auth,
+    currentUserLocale,
+  }: TemplateApiModuleActionProps): Promise<{ success: boolean }> {
+    const fakeData = getValueViaPath<OperatingLicenseFakeData>(
+      application.answers,
+      'fakeData',
+    )
+    const useFakeData = fakeData?.useFakeData === YES
+
+    if (useFakeData) {
+      return fakeData?.debtStatus === YES
+        ? Promise.resolve({ success: true })
+        : Promise.reject({
+            reason: {
+              title: coreErrorMessages.missingCertificateTitle,
+              summary: coreErrorMessages.missingCertificateSummary,
+              hideSubmitError: true,
+            },
+            statusCode: 404,
+          })
+    }
+    const lang = currentUserLocale === 'is' ? 'IS' : 'EN'
+
+    const response = await this.financeService.getDebtLessCertificate(
+      auth.nationalId,
+      lang,
```

Comment:
Not a review comment, just a thought:
Would this benefit from being a map, in the case of more locales in the future?

F.x.

```typescript
const financeServiceLangMap = {
'is': 'IS',
'en': 'EN'
}
//...
//and then

const response = await this.financeService.getDebtLessCertificate(
  auth.nationalId,
  financeServiceLangMap[currentUserLocale] ?? 'is,
  auth
 )
 ```